### PR TITLE
Adds renormalization #7476

### PR DESCRIPTION
**Pull request by shkr**
Adds an additional parameter called renorm to add a stop gradient on gamma, beta parameters during the training phase. This has been discussed here https://github.com/tensorflow/tensorflow/issues/7476

### CODE REVIEW

File: tensorflow/contrib/layers/python/layers/layers.py [hunk 1]
```diff
@@ -135,6 +135,9 @@ def _fused_batch_norm(
     center=True,
     scale=False,
     epsilon=0.001,
+    renorm=False,
+    RMAX=None,
```
Comment (cancan101) [inline tensorflow/contrib/layers/python/layers/layers.py]: `RMAX` and `DMAX` don't have default values and are not documented.

Comment (shkr) [inline tensorflow/contrib/layers/python/layers/layers.py]: I have documented it. According to the paper, it is best initialized by using a scalar tensor which increases w/ the learning step value. However, in the init value I have put in, I have set it to 1.0 or 0.0, which is technically batch normalization. Any suggestions for alternative default tensors ?

File: tensorflow/contrib/layers/python/layers/layers.py [hunk 2]
```diff
@@ -307,8 +313,8 @@ def _fused_batch_norm_training():
           inputs, gamma, beta, epsilon=epsilon, data_format=data_format)
       if renorm:
           moving_inv = math_ops.rsqrt(moving_variance + epsilon)
-          r = tf.stop_gradient(tf.clip_by_value(tf.sqrt(variance + epsilon) * moving_inv, 1/RMAX, RMAX))
-          d = tf.stop_gradient(tf.clip_by_value((mean - moving_mean) * moving_inv, -DMAX, DMAX))
+          r = tf.stop_gradient(tf.clip_by_value(tf.sqrt(variance + epsilon) * moving_inv, 1/r_max, r_max))
```
Comment (cancan101) [inline tensorflow/contrib/layers/python/layers/layers.py]: safer to do `1./r_max`

**Comment by shkr** [comment]
Any updates ?

**Comment by fchollet** [comment]
Sergey Ioffe mentions the following: 

"There are multiple issues with that git change. It keeps maintaining the moving variance instead of standard deviation; will not work early in training due to the lack of debiasing; is inefficient because they do BN first and then correct; and finally the computation is wrong because they apply gamma and beta before the correction, but they should be applied after."

Sergey is working on a correct implementation of batch renormalization and will be pushing it soon. We will not be moving forward with this PR. Sorry about that.
